src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -533,6 +530,9 @@ private void BufferHeaders(HttpRequestMessage request)
                 BufferLiteralHeaderWithStaticNameReference(H3StaticTable.PathSlash, pathAndQuery);
             }
 
+            // The only way to reach H3 is to upgrade via an Alt-Svc header, so we can encode Alt-Used for every connection.
+            BufferBytes(_connection.AltUsedEncodedHeaderBytes);
```

Comment:
This seems like something that could easily get broken in a refactor.  Worth having a test that validates it?

Reply:
Yes, planning to write a bunch of tests soon, I'll create an issue for this one.